### PR TITLE
feat(state): fire state changing as microtask

### PR DESCRIPTION
**Pull request by nattallius**
@exadel-inc/ui-playground-team

### CODE REVIEW

**Review by yadamskaya — APPROVED** [review]

File: src/core/base/model.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ export class UIPStateModel extends Observable {
 
     UIPStateModel.setAttribute(elements, attribute, 'transform' in cfg ? cfg.transform : cfg.value);
     this._lastModifier = modifier;
-    this.fire();
+    Promise.resolve().then(() => this.fire());
```
Comment (ala-n) [inline src/core/base/model.ts:77]: It does not stack into a single call if multiple requests happen per task.

**Review by ala-n — CHANGES_REQUESTED** [review]

File: src/core/base/model.ts (hunk 1 above)
Comment (ala-n) [reply src/core/base/model.ts:77]: The purpose to make something in a microtask is to collect all the changes/requests per task and process the te result change once before the current makrotask ends

**Review by ala-n — APPROVED** [review]

**Review by PolinaVolchetskaya — APPROVED** [review]

**Review by Sisha3342 — APPROVED** [review]

File: src/core/base/model.ts [hunk 2]
```diff
@@ -74,7 +75,17 @@ export class UIPStateModel extends Observable {
 
     UIPStateModel.setAttribute(elements, attribute, 'transform' in cfg ? cfg.transform : cfg.value);
     this._lastModifier = modifier;
-    Promise.resolve().then(() => this.fire());
+    this.promisifiedFiring();
+  }
+
+  protected promisifiedFiring() {
```
Comment (yadamskaya) [inline src/core/base/model.ts:81]: It may be better to use the verb for method name

**Review by AlexanderBazukevich — APPROVED** [review]

**Pull request merged**